index.js

```diff
@@ -1,20 +1,476 @@
+    if (this.grabbable && currentRoom === this.location) {
+      inventory.push(this.name);
+      return `You grabbed the ${this.name}.`;
+      //push item to personal inventory to add it // this part works!
+      //console.log(contents);
+      //console.log(inventory);
+
+      //(filter!==this.name) {
+      // let newContents = content.filter(this.name)
+      // contents.indexOf(this.name);
+      // let pos = contents.indexOf(this.name);
+      //newContents = contents.splice(pos, 1);
+      // console.log(newContents);
+      //}
+    } else if (this.grabbable === false) {  // if an item isn't grabbable it wont go into their inventory and the game will tell them they can't grab it. 
+      return `You can't grab that.\n>_`;
+    }
+    //console.log(this.name);
+    ///let newContents = contents.filter(this.name); // do we need to create a new variable newcontents?
+    // console.log(newContents);
+    //console.log(contents);
+    //let index = contents.indexOf(this.name);
+    //console.log(index);
+    //contents.splice(indexOf(this.name), 0, this.name);
+    //let pos = contents.indexOf(this.name); //indexof is character based?
+    //console.log(pos); // item is not found ....
+    //let removedItem = contents.splice(index); //mayb not splice...
+    //it is not working to remove item from contents
+    //contents.pop(removedItem); //pop item from room contents to remove it
+    //console.log(removedItem);
+    
+  }    // we tried to set it so the contents of a room would drop the item the user puts in their inventory
+
+  use() {
+    if (
+      this.name === "cabinet" ||
+      (this.name === "cabinet" && inventory.includes("studentID"))
+    ) {
+      // if the user has the studentID in their inventory and tries to open the cabinet, it will open.
+      console.log(
+        "you opened the cabinet, to find the lost and found and you drop off all your items here.\n Thank you for playing!"
+      );
+      process.exit();
+    } else {
+      return this.action;
+    } // we need to add each object to one of the actions to put them into play
+  }
+ 
+  examine() {  // if the user examines an item it will return its description 
+    if (this.name === "bubbler") {
+      return bubbler.desc; 
+    } else if (this.name === "globe") {
+      return globe.desc;
+    } else if (this.name === "backpack") {
+      return backpack.desc;
+    } else if (this.name === "folder") {
+      // gives desc of undefined
+      return folder.desc;
+    } else if (this.name === "cabinet") {
+      return cabinet.desc;
+    } else if (this.name === "jacket") {
+      return jacket.desc;
+    } else if (this.name === "hallPass") {
+      return hallpass.desc;
+    } else if (this.name === "steps") {
+      return steps.desc; 
+    } else if (lookupTableOne.includes[this.name] === false) {
+      console.log("Sorry, I can't do that");
+    }  // if the item is not in the lookup table, you can't examine it. 
```

Comment:
This could be refactored to just a single if that says `return this.desc` else `console.log("Sorry I can't do that")`